misc/cf-support

```diff
@@ -0,0 +1,168 @@
+.SH BUGS
+Please see the public bug-tracker at http://bug.cfengine.com/.
+.br
+GitHub pull-requests may be submitted to http://github.com/cfengine/core/.
+.SH "SEE ALSO"
+.BR cf-promises (8),
+.BR cf-agent (8),
+.BR cf-serverd (8),
+.BR cf-execd (8),
+.BR cf-monitord (8),
+.BR cf-runagent (8),
+.BR cf-key (8)
+.SH AUTHOR
+Northern.tech AS
+EOF
+exit 0
+fi
+
+[ "$1" = "--yes" ] || [ "$1" = "-y" ] && non_interactive=1
+# TODO: remove set -x here
+set -e
+if [ "$EUID" -ne 0 ]; then
+  echo "$0 must be run as root"
+  exit 1
+fi
+
+
+if [ $non_interactive -eq 0 ]; then
+  read -p "If you have it, please enter your support case number: " case_number
+fi
+case_number=${case_number:-NA}
+timestamp=$(date +%F-%H%M)
+collection="cfengine_support_case_$case_number-$timestamp"
+
+export tmpdir="$(mktemp -d)/$collection"
+mkdir -p "$tmpdir"
+
+export info_file="$tmpdir/system-info.txt"
+
+function file_add
+{
+  local filename=$(basename $1)
+  if [ -f "$1" ]; then
+    cp "$1" "$tmpdir"/"$filename"
+  else
+    echo "$1 not found" >> "$info_file"
+  fi
+}
+export -f file_add
+
+function gzip_add
+{
+  local filename=$(basename $1)
+  if [ -f "$1" ]; then
+    gzip -c "$1" > "$tmpdir"/"$filename".gz
+  else
+    echo "$1 not found" >> "$info_file"
+  fi
+}
+export -f gzip_add
+
+function log_cmd
+{
+  echo "=== $1 ===" | tee -a "$info_file"
+  eval "$1" | tee -a "$info_file"
+  echo | tee -a "$info_file" # for whitespace easier to read
+}
+export -f log_cmd
+
+log_cmd "hostname -f"
+log_cmd "uname -a"
+[ -f "/etc/os-release" ] && log_cmd "cat /etc/os-release"
+log_cmd "$BINDIR/cf-promises -V"
+log_cmd "free -m"
+log_cmd "df -h"
+log_cmd "ps auwwx"
+log_cmd "top -b -H -c -n1"
+log_cmd "netstat -ie"
+log_cmd "lsof"
+log_cmd "mount"
+file_add "/etc/fstab"
+log_cmd "$BINDIR/cf-key -p $WORKDIR/ppkeys/localhost.pub"
+log_cmd "grep 'version =' $WORKDIR/inputs/promises.cf"
+log_cmd "$BINDIR/cf-key -s"
+log_cmd "$BINDIR/cf-check diagnose"
+command -v systemctl >/dev/null && \
+  log_cmd "systemctl status cfengine3" || \
+  log_cmd "/etc/init.d/cfengine3 status"
+for f in /var/log/CFEngine-Install*; do
+  gzip_add $f
+done
+[ -f /var/log/messages ] && syslog_cmd="cat /var/log/messages"
+[ -f /var/log/syslog ] && syslog_cmd="cat /var/log/syslog"
+[ -z "$syslog_cmd" ] && syslog_cmd="dmesg"
+_syslog_filtered="$tmpdir"/syslog-filtered-for-cfengine.log.gz
+if [ "$(command -v egrep >/dev/null)" ]; then
+  $syslog_cmd | egrep 'cf-|CFEngine' | gzip -c > "$_syslog_filtered"
+else
+  $syslog_cmd | grep cf- | gzip -c > "$_syslog_filtered"
+fi
+gzip_add $WORKDIR/outputs/previous
+file_add $WORKDIR/policy_server.dat
+[ -f $WORKDIR/share/cf-support-nova-hub ] && source $WORKDIR/share/cf-support-nova-hub
+find "$tmpdir"
+ls -l "$tmpdir"
+cat "$info_file"
+
+if [ -f $WORKDIR/share/cf-support-nova-hub ]; then
+  source $WORKDIR/share/cf-support-nova-hub
+fi
+
+tar czf "$collection.tar.gz" --directory $tmpdir/../ . && rm -rf $tmpdir
```

Comment:
I wonder if instead of `find "$tmpdir"` above it might be fancier to add `v` to tar arguments
(and echo a message "creating $collection.tar.gz file" before tar command).

It will serve same purpose (output all files), but also is slightly slower (since tar takes time to compress), and "keeps user entertained" (for a lack of better word) when tar is working.

Reply:
![are-you.gif](https://user-images.githubusercontent.com/202896/184274310-3a5f0ce8-9889-4640-8e8a-925d08cfa02e.gif)

Reply:
The find was debugs for me. I don't think the user will have to wait long. I'll remove the find and keep this as-is.